slackbot/manager.py

```diff
@@ -67,8 +67,14 @@ def get_plugins(self, category, text):
         for matcher in self.commands[category]:
             m = matcher.search(text)
             if m:
+                kwargs = m.groupdict()
+                args = tuple(
+                    m.group(i)
+                    for i in range(1, (m.lastindex or 0) + 1)
+                    if i not in matcher.groupindex.values()
+                )
                 has_matching_plugin = True
-                yield self.commands[category][matcher], to_utf8(m.groups())
+                yield self.commands[category][matcher], to_utf8(args), to_utf8(kwargs)
```

Comment:
`to_utf8` for now can only handle single string or list/tuple/set of strings, so you need to extend it to also handle  dicts whose values are strings.